File: code/sdk/android/cloudsheet-Android-sdk/src/main/kotlin/com/opencloudsheet/response/onedrive/WorkSheetResponses.kt
package com.opencloudsheet.response.onedrive

import com.google.gson.annotations.SerializedName

data class WorkSheetListResponse(
    @SerializedName("value")
    val value: List<WorkSheetData>
)

data class WorkSheetData(
    @SerializedName("id")
    val id: String,

    @SerializedName("name")
    val name: String,

    @SerializedName("position")
    val position: Int? = null
)

data class WorkSheetResponse(
    @SerializedName("id")
    val id: String,

    @SerializedName("name")
    val name: String,

    @SerializedName("position")
    val position: Int? = null
)

data class CreateWorkSheetRequest(
    @SerializedName("name")
    val name: String
)

File: code/sdk/ios/OpenCloudSheet/Sources/OpenCloudSheet/Response/OneDrive/WorkSheetResponses.swift
//
//  WorkSheetResponses.swift
//  OpenCloudSheet
//

import Foundation

struct WorkSheetListResponse: Codable {
    let value: [WorkSheetData]
}

struct WorkSheetData: Codable {
    let id: String
    let name: String
    let position: Int?
}

struct WorkSheetResponse: Codable {
    let id: String
    let name: String
    let position: Int?
}

struct CreateWorkSheetRequest: Codable {
    let name: String
}
